src/fields/multi_field.impl.hh

```diff
@@ -52,6 +52,36 @@ MultiField<spacedim, Value>::MultiField(const MultiField &other)
 
 
 template<int spacedim, class Value>
+MultiField<spacedim,Value> &MultiField<spacedim,Value>::operator=(const MultiField<spacedim,Value> &other)
+{
+	ASSERT( flags().match( FieldFlag::input_copy )  , "Try to assign to non-copy field '%s' from the field '%s'.", this->name().c_str(), other.name().c_str());
+	ASSERT(other.shared_->mesh_, "Must call set_mesh before assign to other field.\n");
+	ASSERT( !shared_->mesh_ || (shared_->mesh_==other.shared_->mesh_),
+	        "Assignment between fields with different meshes.\n");
+
+	// check for self assignement
+	if (&other == this) return *this;
+
+	// class members derived from FieldCommon
+	shared_ = other.shared_;
+    shared_->is_fully_initialized_ = false;
+	set_time_result_ = other.set_time_result_;
+	last_time_ = other.last_time_;
+	last_limit_side_ = other.last_limit_side_;
+	is_jump_time_ = other.is_jump_time_;
+	component_index_ = other.component_index_;
+	this->multifield_ = true;
+
+	// class members of Field class
+	sub_fields_ = other.sub_fields_;
```

Comment:
Not sure this is OK. The assignment for the vector is called which should call copy constructors for individual subfields.
then we copy the name as well for subfields. Please, check this in appropriate unit test.